notification-server/src/bin/service.rs

```diff
@@ -346,6 +354,26 @@ async fn traverse_chain(
     processed_height
 }
 
+async fn catch_up(concordium_client: &mut Client, current_height: AbsoluteBlockHeight, notification_ttl: Duration) -> anyhow::Result<AbsoluteBlockHeight> {
+    let current_block_height = concordium_client
+            .get_consensus_info()
+            .await?
+            .last_finalized_block_height;
+    let lower_bound = AbsoluteBlockHeight {
+        // We are not fast than 2 sec per block, hence this should be conservative enough
+        height: current_block_height.height - notification_ttl.as_secs(),
+    };
+    let lower_bound_time: chrono::DateTime<Utc> = Utc::now() - notification_ttl;
+    let time_ago_block = concordium_client.find_first_finalized_block_no_earlier_than(lower_bound.., lower_bound_time).await?;
+
+    if &time_ago_block.block_height > &current_height {
```

Comment:
This could also just be a max function, however I want to add the logs and metrics